src/Exception/Exception.php

```diff
@@ -56,78 +58,90 @@ class Exception extends \Exception
 	 */
 	private static string $prefix = 'error';
 
-	/**
-	 * Class constructor
-	 *
-	 * @param array|string $args Full option array ('key', 'translate', 'fallback',
-	 *                           'data', 'httpCode', 'details' and 'previous') or
-	 *                           just the message string
-	 */
-	public function __construct(array|string $args = [])
-	{
-		// set data and httpCode from provided arguments or defaults
-		$this->data     = $args['data']     ?? static::$defaultData;
-		$this->httpCode = $args['httpCode'] ?? static::$defaultHttpCode;
-		$this->details  = $args['details']  ?? static::$defaultDetails;
+	public function __construct(
+		array|string $args = [], // @deprecated
+
+		string|null $key = null,
+		array|null $data = null,
+		array|null $details = null,
+		string|null $fallback = null,
+		int|null $httpCode = null,
+		string|null $message = null,
+		Throwable|null $previous = null,
+		bool $translate = true
+	) {
+		$key      ??= $args['key'] ?? null;
+		$fallback ??= $args['fallback'] ?? null;
+		$previous ??= $args['previous'] ?? null;
+
+		$this->data =
+			$data ??
+			$args['data'] ??
+			static::$defaultData;
+
+		$this->httpCode =
+			$httpCode ??
+			$args['httpCode'] ??
+			static::$defaultHttpCode;
+
+		$this->details =
+			$details ??
+			$args['details'] ??
+			static::$defaultDetails;
 
-		// define the Exception key
-		$key = $args['key'] ?? static::$defaultKey;
+		// set the Exception code to the key
+		$this->code = $key ?? static::$defaultKey;
 
-		if (Str::startsWith($key, self::$prefix . '.') === false) {
-			$key = self::$prefix . '.' . $key;
+		if (Str::startsWith($this->code, self::$prefix . '.') === false) {
+			$this->code = self::$prefix . '.' . $this->code;
 		}
 
 		if (is_string($args) === true) {
+			$message ??= $args;
+		}
+
+		if ($message !== null) {
 			$this->isTranslated = false;
```

Comment:
Still needed? Because returns new object.

Reply:
Doesn't really return a new object as `parent::contract()` still modifies the current object. Could also be written as

```php
parent::__construct();
return;
```